apps/api/src/modules/summary/summary.service.ts

```diff
@@ -46,27 +49,92 @@ export class SummaryService extends CoreService<Summary> {
           .on('error', (error) => {
             reject(error);
           });
-      });
+        });
+      }}
+    
+      // async findAllJobs(): Promise<Summary[]> {
+      //   return this.summaryRepository.find({ where: { status: Status.ACTIVE } });
+      // }
+
+   
+    getSummaryById(jobId: number): Promise<Summary[]> {
+        this.logger.log(`Getting summary with id: ${jobId}`);
+        return this.summaryRepository.find({
+          where: {
+            jobId: jobId,
+            status: Status.ACTIVE,
+          },
+        });
     }
-  }
 
-  async findAllJobs(
-    options: QueryOptionsDto): Promise<SummaryResponse> {
-    this.logger.log('Getting all jobs with options.');
 
-    const baseConditions = [
-      { field: 'status', value: Status.ACTIVE },
-    ];
-    const searchableFields = ['createdBy', 'data', 'result'];
 
-    const { items, total } = await super.findAll(
-      options,
-      'summary',
-      searchableFields,
-      baseConditions,
-    );
-    return new SummaryResponse(items, total, options.offset, options.limit);
-  }
+    getPendingSummary(): Promise<Summary[]> {
+        this.logger.log('Getting all active pending summaries');
+        return this.summaryRepository.find({
+          where: {
+            jobStatus: JobStatus.PENDING,
+            status: Status.ACTIVE,
+          },
+        });
+      }
 
-}
 
+    async addSummaryToQueue(): Promise<void> {
+        this.logger.log('Starting CRON job to add pending notifications to queue');
+    
+        if (this.isProcessingQueue) {
+          this.logger.log('Notifications are already being added to queue, skipping this CRON job');
```

Comment:
Fix this, its not notifications.

Reply:
Check for any other places too where its notification

Reply:
Updated